src/compiler/transformers/es2015.ts

```diff
@@ -2583,7 +2583,7 @@ namespace ts {
                     && i < numInitialPropertiesWithoutYield) {
                     numInitialPropertiesWithoutYield = i;
                 }
-                if (property.name!.kind === SyntaxKind.ComputedPropertyName) {
+                if (Debug.checkDefined(property.name).kind === SyntaxKind.ComputedPropertyName) {

```

Comment:
I put this since I figured that it won't hurt to check it, but maybe it's better to use a `?` or maybe leave as is?